snapshot/git/gitdb/db_test.go

```diff
@@ -54,6 +53,58 @@ func TestIngestDir(t *testing.T) {
 	}
 }
 
+func TestIngestCommit(t *testing.T) {
+	id1, err := fixture.db.IngestGitCommit(fixture.external, fixture.commit1ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	id2, err := fixture.db.IngestGitCommit(fixture.external, fixture.commit2ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	co, err := fixture.db.Checkout(id1)
+	if err != nil {
+		t.Fatalf("error checking out %v, %v", id1, err)
+	}
+
+	// test contents
+	data, err := ioutil.ReadFile(filepath.Join(co, "file.txt"))
+	if err != nil || string(data) != "first" {
+		t.Fatalf("error reading file.txt: %q %v (expected \"first\" <nil>)", data, err)
+	}
+
+	// Write temporary data into checkouts to make sure it's cleaned
+	if err = ioutil.WriteFile(filepath.Join(co, "scratch.txt"), []byte("1"), 0777); err != nil {
```

Comment:
Yeah, there's a lot to improve in how we checkout to balance:
*) correctness
*) cleanliness
*) performance